path_login.go

```diff
@@ -273,8 +274,24 @@ func (b *azureAuthBackend) verifyResource(ctx context.Context, subscriptionID, r
 			principalIDs[to.String(vmss.Identity.PrincipalID)] = struct{}{}
 		}
 		// if not, look for user-assigned identities
-		for _, userIdentity := range vmss.Identity.UserAssignedIdentities {
-			principalIDs[to.String(userIdentity.PrincipalID)] = struct{}{}
+		for userIdentityID := range vmss.Identity.UserAssignedIdentities {
```

Comment:
Do you happen to know if it's intentional that the principal ID is omitted for only VMSS Flex? It seems a bit strange that it is missing for Flex but is included for Uniform Orchestration Mode. I'm curious why that might be the case.

Reply:
Yes, it's intentional that the principal ID is omitted for VMSS Flex. This is because this information cannot be directly retrieved from VMSS DB. The Uniform API is designed previously, in which the VMSS needs to query the identity API behind the sense. Now for Flex, to keep the VMSS orchestrator as light, and avoid dependency on another resource, the VMSS engineer is enforcing the API not to query identity any more. But for Uniform, we still do that just for backward compatibility.

Reply:
Thanks for the explanation, @zmyzheng. Makes sense 👍